Result/Result.swift

```diff
@@ -131,6 +131,21 @@ public enum Result<T, Error: ErrorType>: CustomStringConvertible, CustomDebugStr
 	}
 }
 
+public protocol ResultType {
+	typealias Value
+	typealias Error: ErrorType
+	
+	var value: Value? { get }
+	var error: Error? { get }
```

Comment:
Given `analysis`, you can provide `value` and `error` in a protocol extension. That reduces the requirements to two introduction rules (the `init`s) and one elimination rule (`analysis`) covering both alternatives, which is its logical minimum :tada: